lib/scanner/walk.go

```diff
@@ -139,15 +140,20 @@ func (w *walker) walk(ctx context.Context) chan ScanResult {
 	// Parallel hasher is stopped by this routine when we close the channel over
 	// which it receives the files we ask it to hash.
 	go func() {
-		var filesToHash []protocol.FileInfo
-		var total int64 = 1
+		filesToHash := diskoverflow.NewSlice(w.DiskOverflowLocation)
+		defer filesToHash.Close()
 
 		for file := range toHashChan {
-			filesToHash = append(filesToHash, file)
-			total += file.Size
+			filesToHash.Append(file)
+		}
+
+		if filesToHash.Items() == 0 {
+			close(finishedChan)
+			return // nothing to do
 		}
 
-		realToHashChan := make(chan protocol.FileInfo)
+		total := filesToHash.Bytes()
```

Comment:
That's not the right total, this was counting the total size of files that are represented by the index, this now shows the size of the index O_o